config/application.rb

```diff
@@ -153,7 +153,14 @@ class Application < Rails::Application
 
     # enables in memory cache store with ttl
     # config.cache_store = TimedCachedStore.new
-    config.cache_store = :file_store, Rails.root.join("tmp", "cache")
+    if SETTINGS[:cache_store] == 'redis'
+      args = [:redis_cache_store]
+      Array.wrap(SETTINGS[:redis_hosts]).each { |h| args << h }
+      args << { namespace: 'foreman' }.merge(SETTINGS[:redis_cache_options] || {})
```

Comment:
could we add support to define it via ENV variable as well? something like
```rb
config.cache_store = :redis_cache_store, { url: ENV['REDIS_URL'] }
```

Reply:
another option, would be to allow defining the entire yaml as a env variable, something like
```diff
diff --git a/config/settings.rb b/config/settings.rb
index 7640ed7e5..45141b3da 100644
--- a/config/settings.rb
+++ b/config/settings.rb
@@ -5,6 +5,7 @@ root = File.expand_path(File.dirname(__FILE__) + "/..")
 settings_file = Rails.env.test? ? 'config/settings.yaml.test' : 'config/settings.yaml'
 
 SETTINGS.merge! YAML.load(ERB.new(File.read("#{root}/#{settings_file}")).result) if File.exist?(settings_file)
+SETTINGS.merge! YAML.load(ENV['FOREMAN_SETTINGS']) if ENV['FOREMAN_SETTINGS']
 SETTINGS[:version] = Foreman::Version.new
 
```
WDYT?

Reply:
I don't think a YAML settings hash is a great idea since it's very hard to manipulate.

How about shipping a `config/settings.yaml` file inside the docker environment (content completely made up):
```erb
---
:redis_cache_store: <%= ENV['REDIS_CACHE_STORE'] || 'my_store' %>
```

Reply:
That would be the easiest solution, but I think we should be able to parse the ENV variables as well and store them in a settings hash. Ideally, we would also validate the boot settings file to make sure only allowed variables are set.

Reply:
I've filed a PR at https://github.com/theforeman/foreman/pull/6826 that adds proper support for defining settings in env variables.